[abenx162] iP

abenx162 opened the pull request:
# Page, the medieval chatbot!
- Are you looking for an app to keep track of your tasks?
- Do you want to have fun while doing it?

Use Page! Features:
1. add and delete Quests to your Quest Log
2. search Quest Log for keywords
3. and more!

Here's a behind-the-scenes look:
```Java
    public static void main(String[] args) {
        Page page = new Page("data/questlog.txt");
        page.run();
    }
```

Keep track of tasks like this!
[ ] task 1
[x] task 2
[ ] task 3

Look at this review 👍 

> "I love using Page!" - Anonymous

Try it out [here](https://github.com/abenx162/ip/releases/tag/v0.1)!
hint: while using Page, type `help` to see a list of commands.

**THANK YOU** for reading!

Commit:
Add Gradle support

abenx162 committed:
19/01/23 Level-1: Greet, Echo, Exit

abenx162 committed:
19/01/23 Level-2: Add, List +help function

abenx162 committed:
19/01/23 Level-3 Mark as Done, A-Classes

abenx162 committed:
19/01/23 renamed 'list' to 'log'

abenx162 committed:
19/01/23 Level-4, A-Inheritance

abenx162 committed:
20/01/23 A-TextUiTesting

abenx162 committed:
25/01/23 Level-5, A-Exceptions

abenx162 committed:
25/01/23 Level-6

abenx162 committed:
27/01/23 branch-level-7

abenx162 committed:
31/01/23 branch-Level-8

abenx162 committed:
Merge branch 'branch-Level-8'

abenx162 committed:
Merge branch 'branch-Level-7'

wxxedu commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Quest {
+    private LocalDateTime by;
```
Would it be better if we name this variable as `finishBy`?

wxxedu commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,21 @@
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Quest {
+    private LocalDateTime from;
```
Can we change this to `fromDate` and `toDate` just so that the variable names could be more explicit?

wxxedu commented on src/main/java/Page.java:
```diff
@@ -0,0 +1,222 @@
+
+public class Page {
+
+    private Scanner scan;
```
Would it be better if we use `scanner` for the variable name? I've noticed similar issues in many other places as well.

wxxedu reviewed (commented):
I am wondering if you can add some doc comments as well. Also, I think it would be better to change some names in the code.

abenx162 committed:
01/02/23 A-MoreOOP

abenx162 committed:
01/02/23 A-Packages

abenx162 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

bofeng1999 commented on src/main/java/page/quest/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+
+public class Deadline extends Quest {
+    private LocalDateTime by;
+    private static DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("[HHmm dd/MM/yy][dd MMM yyyy hh:mma]");
```
This line may be abit too long, break it up into two lines if necessary

bofeng1999 commented on src/main/java/page/Storage.java:
```diff
@@ -0,0 +1,69 @@
+        this.storageFile = new File(filePath);
+    }
+
+    public ArrayList<Quest> loadData() throws PageException {
```
Maybe add JavaDocs header comments, does not seems like a trivial method

bofeng1999 commented on src/main/java/page/Parser.java:
```diff
@@ -0,0 +1,95 @@
+
+public class Parser {
+
+    public boolean parseExecute(String input, Ui ui, Storage storage, QuestLog questLog) throws PageException {
```
Add abstraction for this method, seems a bit too long

bofeng1999 commented on src/main/java/page/Parser.java:
```diff
@@ -0,0 +1,95 @@
+                                throw new PageException("Sorry, please include a /from time!");
+                            } else {
+                                String[] splitTo = splitFrom[1].split(" /to ");
+                                if (splitTo.length == 1) {
```
Very deep nesting here, a bit hard to read, maybe can extract out certain code as a different method

bofeng1999 reviewed (commented):
LGTM! Just some minor improvements needed! :)

abenx162 committed:
03/02/23 A-Gradle

abenx162 committed:
03/02/23 A-JUnit

abenx162 committed:
03/02/23 A-CodingStandard

abenx162 committed:
03/02/23 Removed JavaDocs to put in another branch.

abenx162 committed:
03/02/23 A-JavaDoc

abenx162 committed:
03/02/23 Level-9

abenx162 committed:
Merge branch 'branch-A-JavaDoc'

abenx162 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/page/Page.java
#	src/main/java/page/QuestLog.java
#	src/main/java/page/Storage.java
#	src/main/java/page/Ui.java
#	src/main/java/page/quest/Deadline.java
#	src/main/java/page/quest/Event.java
#	src/main/java/page/quest/Quest.java

abenx162 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/page/Parser.java
#	src/main/java/page/QuestLog.java
#	src/main/java/page/Ui.java
#	src/main/java/page/quest/Quest.java

abenx162 committed:
Add files for GUI

A template for setting up the GUI for Page is needed.

Let's add the relevant classes needed for the JavaFX GUI, as well as the necessary .fxml files.

abenx162 committed:
Add images for GUI

We need icons to represent the user and Page.

Let's add these images so the GUI can access them.

abenx162 committed:
Extract Command classes

abenx162 committed:
Finish GUI

abenx162 committed:
Merge branch 'branch-Level-10'

abenx162 committed:
Change build.gradle such that .jar file will properly run the Launcher

abenx162 committed:
Add assertions

abenx162 committed:
Minor code quality changes

abenx162 committed:
Merge pull request #2 from abenx162/branch-A-Assertions

Add assertions

abenx162 committed:
Merge pull request #3 from abenx162/branch-A-CodeQuality

Minor code quality changes

abenx162 committed:
Add Edit Quest function

abenx162 committed:
Merge branch 'branch-BCD-Extension'

abenx162 committed:
Add User Guide and Ui.png

abenx162 committed:
Add some documentation
